R/add_labels.R

```diff
@@ -0,0 +1,157 @@
+#' @title Add variable and value labels
+#' @name add_labels
+#'
+#' @description
+#' Add variable and values labels to a variable or variables in a data frame.
+#' Labels are stored as attributes (`"label"` for variable labels and `"labels"`)
+#' for value labels.
+#'
+#' @param x A data frame, factor or vector.
+#' @param variable The variable label as string.
+#' @param values The value labels as (named) character vector. If `values` is
+#' *not* a named vector, the length of lables must be equal to the length of
+#' unique values. For a named vector, the left-hand side (LHS) is the value in
+#' `x`, the right-hand side (RHS) the associated value label. Non-matching
+#' labels are omitted.
+#' @param ... Currently not used.
+#' @inheritParams find_columns
+#'
+#' @inheritSection center Selection of variables - the `select` argument
+#'
+#' @return A labelled variable, or a data frame of labelled variables.
+#'
+#' @examples
+#' x <- 1:3
+#' # labelling by providing required number of labels
+#' add_labels(
+#'   x,
+#'   variable = "My x",
+#'   values = c("one", "two", "three")
+#' )
+#'
+#' # labelling using named vectors
+#' data(iris)
+#' out <- add_labels(
+#'   iris$Species,
+#'   variable = "Labelled Species",
+#'   values = c(`setosa` = "Spec1", `versicolor` = "Spec2", `virginica` = "Spec3")
+#' )
+#' str(out)
```

Comment:
The names and values of `attr("labels")` are inverted compared to `sjlabelled::add_labels()`:
``` r
df <- iris[c(1, 55, 108), ]

datawizard::add_labels(
  df$Species,
  variable = "Labelled Species",
  values = c(`setosa` = "Spec1", `versicolor` = "Spec2", `virginica` = "Spec3")
)
#> [1] setosa     versicolor virginica 
#> attr(,"label")
#> [1] Labelled Species
#> attr(,"labels")
#>      Spec1      Spec2      Spec3 
#>     setosa versicolor  virginica 
#> Levels: setosa versicolor virginica

sjlabelled::add_labels(
  df$Species,
  labels = c(`setosa` = "Spec1", `versicolor` = "Spec2", `virginica` = "Spec3")
)
#> Warning in order(as.numeric(all.labels)): NAs introduced by coercion
#> [1] setosa     versicolor virginica 
#> attr(,"labels")
#>     setosa versicolor  virginica 
#>      Spec1      Spec2      Spec3 
#> Levels: setosa versicolor virginica
```

Reply:
No, it's handled internally, and I wanted to follow the pattern `"value = label"`.  in `sjlabelled::add_labels()`, it is `"label = value"`. What do you think makes more sense? The implementation here follows the data structure from `haven::labelled()`.

Reply:
I meant in the output, the names and the values of the "labels" vector are inverted.

In the function call, I personally prefer `"value = label"` but maybe it's better to keep the order in which `sjlabelled` does it to reduce the cognitive load

Reply:
They're inverted because the correct call to that function would be

```r
sjlabelled::add_labels(
  df$Species,
  labels = c("Spec1" = "setosa", "Spec2" = "versicolor", "Spec3" = "virginica")
)
```

> to reduce the cognitive load

We already have this in `recode_values()`, which has the opposite pattern as compared to `sjmisc::rec()`. I'd say we can have a "breaking change" here.

Reply:
Ah ok, it's clearer now, thanks. 

True and `dplyr::recode()` also does `old = new`. What is best to keep consistency both inside `datawizard` and with other packages? 

Speaking of `recode_values()`, I have 2 questions: 1) why do we need a list in the second arg while `recode_into()` works fine with the dots? 2) do we need both `recode_values()` and `recode_into()`?

Reply:
First, `recode_values()` is a generic, where dots are somewhat "reserved". Second, not necessarily. `recode_values()` is a bit shorter and maybe more "intuitive" or clearer for the case of recoding just one variable. But yes, I think you can achieve all recodings you could do in `recode_values()` with `recode_into()`, as well.